lib/portage/dbapi/vartree.py

```diff
@@ -5703,6 +5703,15 @@ def mergeme(
                     ):
                         # a symlink to an existing directory will work for us; keep it:
                         showMessage(f"--- {mydest}/\n")
+                        # update owner and mode on existing directories
+                        if stat.S_ISDIR(mydmode):
+                            if (
+                                mydstat.st_uid != mystat.st_uid
+                                or mydstat.st_gid != mystat.st_gid
+                            ):
+                                os.chown(mydest, mystat[4], mystat[5])
```

Comment:
this and the later `chown` can probably be done unconditionally, maybe even for symlinks

the code is already pretty wild and hard to read

but for now i opted for a conservative way where addition warning prints could be added

Reply:
should be rebased on top of https://github.com/gentoo/portage/pull/1382 and not use magic numbers